src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,139 @@ def standardize_patient_name(
     return bundle
 
 
-def standardize_phone(raw: str) -> str:
-    """Make sure it's 10 digits, remove everything else
+def country_extractor(
+    resource: dict, code_type: Literal["alpha_2", "alpha_3", "numeric"] = "alpha_2"
+) -> Generator[str, None, None]:
+    """
+    Given a FHIR resource yield a generator containing all of the counries found in the
+    resource in a standard form sepcified by code_type.
+
+    Args:
+        resource (dict): A FHIR resource respresented as a dictionary.
+        code_type (str): A string equal to 'alpha_2', 'alpha_3', or 'numeric' to specify
+                         which type of standard country identifier to generate.
+    Returns:
+        (generator): A generator to return standardized countries when it is iterated
+                     over.
+
+    We currently only handle extracting countries from patient resources, however as
+    need arises is the future funcionality can be expanded.
+    """
+    resource_type = resource.get("resource").get("resourceType")
+    if resource_type == "Patient":
+        for address in resource.get("resource").get("address"):
```

Comment:
It seems unlikely to me that any given patient resource will have so many addresses that maintaining a list of the countries will be an expensive task requiring a generator. I like this approach, but by just getting a list of countries, you can call this function outside of `standardize_phone` and pass in the list of countries instead of the entire resource.

Reply:
Seconded. I think at this point we really need to be emphasizing human-readability, uniformity in design, and composability (can you tell I'm deep in the refactors lol). I don't think we need to provide a generator to the list of countries and think a list that could be accessed outside `standardize_phone` would be much better.

Reply:
I agree, I was looking for away to only actually loop over the addresses in a patient resource once and opposed to twice. The loops won't be expensive and I agree it would clean up other elements of the code.